editing/edit-with-branch-versioning/src/main/java/com/esri/samples/edit_with_branch_versioning/EditWithBranchVersioningController.java

```diff
@@ -0,0 +1,352 @@
+
+      // add the version access types to the combo box
+      accessComboBox.getItems().addAll(VersionAccess.PUBLIC, VersionAccess.PROTECTED, VersionAccess.PRIVATE);
+
+      // add the damage types to the combo box and handle selection
+      damageTypeComboBox.getItems().addAll("Destroyed", "Inaccessible", "Major", "Minor", "Affected");
+      damageTypeComboBox.getSelectionModel().selectedItemProperty().addListener((o, p, n) -> {
+        if (!selectedFeature.getAttributes().get("typdamage").equals(n)) {
+          try {
+            selectedFeature.getAttributes().put("typdamage", damageTypeComboBox.getValue());
+            updateFeature(selectedFeature);
+          } catch (Exception e) {
+            new Alert(Alert.AlertType.ERROR, "Feature attributes failed to update").show();
+          }
+        }
+      });
+
+      // handle authentication for the service geodatabase
+      AuthenticationManager.setAuthenticationChallengeHandler(new DefaultAuthenticationChallengeHandler());
+
+      // create and load a service geodatabase
+      serviceGeodatabase = new ServiceGeodatabase("https://sampleserver7.arcgisonline" +
+        ".com/arcgis/rest/services/DamageAssessment/FeatureServer");
+      serviceGeodatabase.loadAsync();
+      serviceGeodatabase.addDoneLoadingListener(() -> {
+        if (serviceGeodatabase.getLoadStatus() == LoadStatus.LOADED) {
+
+          // when the service geodatabase has loaded get the default version
+          defaultVersion = serviceGeodatabase.getDefaultVersionName();
+
+          // get the service feature table from the service geodatabase
+          if (serviceGeodatabase.getTable(0) != null) {
+            serviceFeatureTable = serviceGeodatabase.getTable(0);
+
+            // create a feature layer from the service feature table and add it to the map
+            featureLayer = new FeatureLayer(serviceFeatureTable);
+            map.getOperationalLayers().add(featureLayer);
+            featureLayer.addDoneLoadingListener(() -> {
+              if (featureLayer.getLoadStatus() == LoadStatus.LOADED) {
+
+                // when the feature layer has loaded set the viewpoint and update the UI
+                mapView.setViewpointAsync(new Viewpoint(featureLayer.getFullExtent()));
+                progressIndicator.setVisible(false);
+                createVersionButton.setDisable(false);
+                currentVersionLabel.setText("Current version: " + serviceGeodatabase.getVersionName());
+              } else {
+                new Alert(Alert.AlertType.ERROR, "Feature layer failed to load").show();
+              }
+            });
+          } else {
+            new Alert(Alert.AlertType.ERROR, "Unable to get the service feature table").show();
+          }
+        } else {
+          progressIndicator.setVisible(false);
+          new Alert(Alert.AlertType.ERROR, "Service geodatabase failed to load").show();
+        }
+      });
+
+      mapView.setOnMouseClicked(event -> {
```

Comment:
Let's add a comment here to tell the users roughly what's going on in the block, something along the lines of :
```suggestion
	  // listen to clicks on the map to select or move features
      mapView.setOnMouseClicked(event -> {
```